x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -620,7 +620,10 @@ export class AlertsClient {
       }
 
       // move away from pulling data from _source in the future
-      return alert.hits.hits[0]._source;
+      return {
+        _index: alert.hits.hits[0]._index,
+        ...alert.hits.hits[0]._source,
+      } as ParsedTechnicalFields & { _index?: string };
```

Comment:
I noticed that the `_index` will always be part of the `alert.hits.hits[0]`. Should we make it required instead of optional? Also what about creating a dedicated type for the response?

Reply:
@cnasikas I have refactored the code to address type issues. I am now sending `_index` and `_source` as separate fields in [response](https://github.com/elastic/kibana/pull/181297/files#diff-2e34469c24282f59b0a2eae766b59c048ddc07b608201b96bad35641bf029bb5R623-R626).

Reply:
you are right, changing schema could cause other issues. I have reverted the code to earlier approach where response will include `_index` at the same level as other fields.